### PR TITLE
test(datastore): Flutter update

### PR DESCRIPTION
**Pull request by mikepschneider**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*
https://app.asana.com/0/1202377353696664/1202489222481362/f

*Description of changes:*
When converting TIME and DATE_TIME types for flutter, do not format them, just pass the string through.

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x ] Added Unit Tests
- [ x] Tested on sample app
- [ x] Successful logs (Attach them to the PR)

*Documentation update required?*
- [x ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Successful log:
```I/flutter (18643): [ModelWithAppsyncScalarTypes {id=51290c77-94ac-4a29-b451-7f0629e2c640, stringValue=null, altStringValue=null, listOfStringValue=null, intValue=null, altIntValue=null, listOfIntValue=null, floatValue=null, listOfFloatValue=null, booleanValue=null, listOfBooleanValue=null, awsDateValue=null, listOfAWSDateValue=null, awsTimeValue=16:00:00.050020000, listOfAWSTimeValue=null, awsDateTimeValue=null, listOfAWSDateTimeValue=null, awsTimestampValue=null, listOfAWSTimestampValue=null, awsEmailValue=null, listOfAWSEmailValue=null, awsJsonValue=null, listOfAWSJsonValue=null, awsPhoneValue=null, listOfAWSPhoneValue=null, awsURLValue=null, listOfAWSURLValue=null, awsIPAddressValue=null, listOfAWSIPAddressValue=null, createdAt=null, updatedAt=null}, ModelWithAppsyncScalarTypes {id=71c0e79e-c58b-4793-ab90-67665366ea1a, stringValue=null, altStringValue=null, listOfStringValue=null, intValue=null, altIntValue=null, listOfIntValue=null, floatValue=null, listOfFloatValue=null, booleanValue=null, listOfBooleanValue=null, awsDate```

### CODE REVIEW

File: aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelFieldTypeConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * Test DATE_TIME converter for Flutter.
+     * @throws AmplifyException Not expected.
+     */
+    @Test
+    public void testConvertRawValueToTargetDateTimeFlutter() throws AmplifyException {
+        UserAgent.configure(Map.of(UserAgent.Platform.FLUTTER, "1.0"));
+        final String value = "16:00:00.050020000";
```
Comment (HuiSF) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelFieldTypeConverterTest.java:82]: nit: Can use a date time string as it's a test for AWSDateTime value. e.g. `2020-01-01T16:00:00.050020000`

**Review by HuiSF — DISMISSED** [review]
Thanks for fixing this issue.

**Review by HuiSF — APPROVED** [review]

**Review by poojamat — APPROVED** [review]

**Review by drochetti — APPROVED** [review]

**Review by div5yesh — DISMISSED** [review]
LGTM

**Review by div5yesh — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Comment by HuiSF** [comment]
This PR fixes this issue: https://github.com/aws-amplify/amplify-android/issues/1736

**Pull request merged**